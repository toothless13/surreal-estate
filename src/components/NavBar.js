import React, { useContext } from "react";
import { Link } from "react-router-dom";
import "../styles/navbar.css";
import logo from "../styles/resources/logo.png";
import Logout from "./Logout";
import { Context } from "../Context/AuthContext";

const NavBar = () => {
  const { user } = useContext(Context);

  return (
    <div className="navbar">
      <img
        className="logo"
        src={logo}
        alt="surreal estate logo with small castle"
      />
      <ul className="navbar-links">
        <li className="navbar-links-item">
          <Link to="/">View Properties</Link>
        </li>
        <li className="navbar-links-item">
          <Link to="add-property">Add a Property</Link>
        </li>
        {user && (
          <li className="navbar-links-item">
            <Link to="saved-properties">Saved Properties</Link>
          </li>
        )}
      </ul>
      <ul className="account-links">
        {!user ? (
          <li className="navbar-links-item navbar-login">
            <Link to="login">Login</Link>
          </li>
        ) : (
          <li className="navbar-links-item navbar-logout">
            <Logout />
          </li>
        )}
      </ul>
    </div>
  );
};

export default NavBar;
