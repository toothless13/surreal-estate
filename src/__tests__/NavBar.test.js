import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import NavBar from "../components/NavBar";

describe("NavBar", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>,
    );

    const logo = screen.getAllByAltText(
      /surreal estate logo with small castle/i,
    );
    const firstLink = screen.getByText(/View Properties/i);
    const secondLink = screen.getByText(/Add a Property/i);
    
    expect(asFragment()).toMatchSnapshot();
    expect(logo).toBeTruthy();
    expect(firstLink).toBeTruthy();
    expect(secondLink).toBeTruthy();
  });
});
