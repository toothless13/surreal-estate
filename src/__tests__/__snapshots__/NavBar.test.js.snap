// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <img
      alt="surreal estate logo with small castle"
      class="logo"
      src="logo.png"
    />
    <ul
      class="navbar-links"
    >
      <li
        class="navbar-links-item"
      >
        <a
          href="/"
        >
          View Properties
        </a>
      </li>
      <li
        class="navbar-links-item"
      >
        <a
          href="/add-property"
        >
          Add a Property
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
